torch/csrc/utils/tensor_new.cpp

```diff
@@ -146,4 +159,25 @@ Tensor tensor_new(const Type& type, PyObject* args, PyObject* kwargs) {
   throw std::runtime_error("new(): invalid arguments");
 }
 
+static const Tensor & set_requires_grad(const Tensor &self, bool requires_grad) {
```

Comment:
Minor but I think the signature should be:

`static Tensor set_requires_grad(Tensor self, bool requires_grad)`

So that the move instead of copy constructor is called.